src/Propulsion.MessageDb/MessageDbSource.fs

```diff
@@ -0,0 +1,97 @@
+namespace Propulsion.MessageDb
+
+open System
+open System.Data.Common
+open System.Diagnostics
+open FSharp.Control
+open FsCodec
+open FsCodec.Core
+open Npgsql
+open NpgsqlTypes
+open Propulsion.Feed
+
+type MessageDbCategoryReader(connectionString) =
+    let readonly (bytes: byte array) = ReadOnlyMemory.op_Implicit(bytes)
+    let readRow (reader: DbDataReader) =
+        let readNullableString idx = if reader.IsDBNull(idx) then None else Some (reader.GetString idx)
+        let timestamp = DateTimeOffset(DateTime.SpecifyKind(reader.GetDateTime(7), DateTimeKind.Utc))
+        let streamName = reader.GetString(8)
+        let event =TimelineEvent.Create(
+           index = reader.GetInt64(0),
+           eventType = reader.GetString(1),
+           data = (reader.GetFieldValue<byte array>(2) |> readonly),
+           meta = (reader.GetFieldValue<byte array>(3) |> readonly),
+           eventId = reader.GetGuid(4),
+           ?correlationId = readNullableString 5,
+           ?causationId = readNullableString 6,
+           timestamp = timestamp)
+
+        struct(StreamName.parse streamName, event)
+    member _.ReadCategoryMessages(category: TrancheId, fromPositionInclusive: int64, batchSize: int, ct) = task {
+        use conn = new NpgsqlConnection(connectionString)
```

Comment:
Maybe have the ctor take a `createConnection : unit -> NpgsqlConnection`  ?
that'd take out the connection string/logic dep, and also make the create/open be DRY